semgrep-core/src/parsing/other/yaml_to_generic.ml

```diff
@@ -224,6 +224,7 @@ let make_pattern_expr e =
   (* If the user creates a pattern with a single field, assume they just want *
    * to match the field, not the whole enclosing container                    *)
   | A.Container(A.Dict, (_lp, [x], _rp)) -> A.E x
+  | A.Container(A.Array, (_lp, [x], _rp)) -> A.E x
```

Comment:
However this will match anything, not just in an Array context. For Dict it was ok, because the
element was a Tuple(key,val) which can appear only in a Dict, but for Array it's not the case.

Reply:
hm should we make some sort of distinction then?

Reply:
let's try like this, see if people complain :)